packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -469,48 +489,170 @@ class HeapTreeViewState extends State<HeapTree>
+                            setState(() {});
+                          }
                         }
-                      }
-                    : null,
-                child: const Text('Expand All'),
+                      : null,
+                  child: const MaterialIconLabel(
+                    Icons.vertical_align_bottom,
+                    '',
+                  ),
+                ),
               ),
+        const SizedBox(width: defaultSpacing),
+
+        Tooltip(
+          message: 'Monitor Allocations',
+          child: OutlineButton(
+            key: allocationMonitorKey,
+            onPressed: () async {
+              await _allocationStart();
+            },
+            child: createImageIcon(
+              'icons/memory/communities_white@2x.png',
+              size: defaultIconThemeSize,
+            ),
+          ),
+        ),
+        Tooltip(
+          message: 'Reset Accumulators',
+          child: OutlineButton(
+            key: allocationMonitorResetKey,
+            onPressed: () async {
+              await _allocationReset();
+            },
+            child: createImageIcon(
+              'icons/memory/reset_icon_white@2x.png',
+              size: defaultIconThemeSize,
+            ),
+          ),
+        ),
       ],
     );
   }
 
+  Future<void> _allocationStart() async {
+    // TODO(terry): Look at grouping by library or classes also filtering e.g., await controller.computeLibraries();
+    controller.memoryTimeline.addMonitorStartEvent();
+
+    final allocationtimestamp = DateTime.now();
+    final currentAllocations = await controller.getAllocationProfile();
+
+    if (controller.monitorAllocations.isNotEmpty) {
+      final previousLength = controller.monitorAllocations.length;
+      int previousIndex = 0;
+      final currentLength = currentAllocations.length;
+      int currentIndex = 0;
+      while (currentIndex < currentLength && previousIndex < previousLength) {
+        final previousAllocation = controller.monitorAllocations[previousIndex];
+        final currentAllocation = currentAllocations[currentIndex];
+
+        if (previousAllocation.classRef.id == currentAllocation.classRef.id) {
+          final instancesCurrent = currentAllocation.instancesCurrent;
+          final bytesCurrent = currentAllocation.bytesCurrent;
+
+          currentAllocation.instancesAccumulated =
+              previousAllocation.instancesAccumulated +
+                  (instancesCurrent - previousAllocation.instancesCurrent);
+          currentAllocation.bytesAccumulated =
+              previousAllocation.bytesAccumulated +
+                  (bytesCurrent - previousAllocation.bytesCurrent);
+
+          final instancesAccumulated = currentAllocation.instancesAccumulated;
+          final bytesAccumulated = currentAllocation.bytesAccumulated;
+
+          if (instancesAccumulated != 0 || bytesAccumulated != 0) {
+            debugLogger('previous,index=[$previousIndex][$currentIndex] '
```

Comment:
should this logging be removed? this seems like it is logging a ton of data given the if condition is usually true. if this is an exceptional case, explain what worrying condition occurred.

Reply:
I've added this under a _debugAllocationMonitoring flag.  May remove after I'm happy that the monitoring is working well.